.github/workflows/typescript-typecheck.yml

```diff
@@ -0,0 +1,24 @@
+name: CI - Node.js Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  type-check:
+    name: Check Typescript typings
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    steps:
+      - name: Checkout source files
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm run type-check
```

Comment:
Since this and node.js-test.yml share a lot of setup and are doing essentially the same thing, I think it's easier to just have the one workflow.

So, I recommend adding one extra step to the other workflow with this command